feathr_project/test/test_azure_spark_e2e.py

```diff
@@ -393,6 +393,66 @@ def test_feathr_materialize_to_aerospike():
     client.materialize_features(settings)
     # assuming the job can successfully run; otherwise it will throw exception
     client.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
+
+def test_feathr_materialize_with_time_partition_pattern():
+    """
+    Test FeathrClient() using HdfsSource with 'timePartitionPattern'.
+    """
+    test_workspace_dir = Path(
+        __file__).parent.resolve() / "test_user_workspace"
+    # os.chdir(test_workspace_dir)
+    # Create data source first
+    client_producer: FeathrClient = basic_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"))
+
+    backfill_time = BackfillTime(start=datetime(
+        2020, 5, 20), end=datetime(2020, 5, 20), step=timedelta(days=1))
+
+    if client_producer.spark_runtime == 'databricks':
+        output_path = 'dbfs:/timePartitionPattern_test'
+    else:
+        output_path = 'abfss://feathrazuretest3fs@feathrazuretest3storage.dfs.core.windows.net/timePartitionPattern_test'
+    
+    offline_sink = HdfsSink(output_path=output_path)
+    settings = MaterializationSettings("nycTaxiTable",
+                sinks=[offline_sink],
+                feature_names=[
+                    "f_location_avg_fare", "f_location_max_fare"],
+                backfill_time=backfill_time)
+    client_producer.materialize_features(settings)
+    # assuming the job can successfully run; otherwise it will throw exception
+    client_producer.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
+
+    # download result and just assert the returned result is not empty
+    # by default, it will write to a folder appended with date
+    res_df = get_result_df(client_producer, "avro", output_path + "/df0/daily/2020/05/20")
+    assert res_df.shape[0] > 0
+    
```

Comment:
Can we add a bit comment here on how is this test being setup? Looks like what we are doing is try to generate a bunch of files in a certain folder with some path patterns, and read those files?

Reply:
Correct. Since we need to make sure there're some data sources under the paths we want to test (eg. basePath + {yyyy/mm/dd}), so create these data sources in advance by running a materialize job for both synapse and databricks.